main.tf

```diff
@@ -32,16 +97,22 @@ module "obs_staging_vpc" {
   deploy_ec2_instance_db                 = false
   deploy_rds_db                          = true
   xchain_settings = {
-    enabled      = true
-    docker_image = local.xchain_indexer_staging_docker_image
-    config       = "staging"
-    omni_config = {
-      omni_rpc = local.omni_staging_rpc
+    enabled             = true
+    docker_image        = local.xchain_indexer_staging_docker_image
+    config_file_name    = "config"
+    config_file_content = jsonencode({
+      omni_config = local.omni_chain_config_staging,
+      external_chains = [
+        for x_chain in local.external_chains_staging : x_chain
+      ]
+    })
+    omni_config      = {
```

Comment:
Kept for now because i am unsure if the image of xchain indexer that expects omni-rpc flag has already been deployed to testnet. In case that it was, this makes sure we do not break anything while testing the mount config approach on staging. As soon as it is tested, this omni_config block would be completely removed.